parsing.js

```diff
@@ -86,15 +95,181 @@ function* verseGenerator(document, verses) {
         console.log("Verse Location: ", verse_location);
         verse.number = v + 1;
         verse.body_index = verses[verse_ids[v]];
-        verse.linguistics = parseLinguistics(document, verse_location);
-        verse.tafsir = parseTafsir(document, verse_location);
-        verse.comments = parseComments(document, verse_location);
+        verse.linguistics = {};
+        verse.variantReadings = {};
+        verse.existingCommentary = {};
+        verse.comments = {};
+        verse.connections = {};
+        parseVerse(document, verse, verse_location + 1);
         console.log(verse);
 
         yield verse;
     }
 }
 
+/**
+ * A function that focuses on parsing all the sections of the current verse
+ *
+ * @param {Object} document
+ * @param {Object} verse
+ * @param {int} verse_loc
+ */
+function parseVerse(document, verse, verse_loc) {
+    let index = verse_loc;
+    let content = document.body.content;
+    let line = content[verse_loc];
+
+    while (index < content.length && !(line?.table && line.table.tableRows[0].tableCells[0].tableCellStyle?.backgroundColor?.color?.rgbColor?.green > 0.9)) {
+        line = content[index];
+
+        // start of a verse section
+        if (line?.paragraph?.elements[0]?.textRun?.textStyle?.underline) {
+            let verse_header = line?.paragraph?.elements[0].textRun.content.split(":")[0];
+            if (headings.includes(verse_header)) index = fnMap[verse_header](document, verse, index);
+        }
+
+        index++;
+    }
+}
+
+// TODO - this feels very repeated (follow DRY principle)
```

Comment:
i left this comment here, but the reason I still have not changed it and kept it this way is because in case we want to parse each section differently down the line (for example, Linguistics section has subnotes for each main note/point - maybe at some point we want to parse each subnote and main note differently). If that won't happen, then I could try to make the code less repetitive